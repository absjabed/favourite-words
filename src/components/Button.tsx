import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import Colors from '../constants/Colors/Colors';
const AppButton = (props: any) => {
  return (
    <TouchableOpacity style={{...styles.button, ...props.style}} disabled={props.disabled} onPress={props.onPress}>
      <Text style={styles.text}>{props.title}</Text>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    backgroundColor: Colors.secondary,
    borderRadius: 7,
    color: Colors.white,
    padding: 10,
    marginTop: 16,
    marginLeft: 20,
    marginRight: 20,
  },
  text: {
    color: Colors.white,
    fontWeight: 'bold'
  },
});
export default AppButton;