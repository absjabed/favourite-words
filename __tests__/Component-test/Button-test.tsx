import 'react-native';
import React from 'react';
import Button from '../../src/components/Button';
import renderer, { create } from 'react-test-renderer';

it('Button renders correctly', () => {
  renderer.create(<Button />);
});

test('Button snapshot', () => {
  const tree = create(<Button />);
  expect(tree).toMatchSnapshot();
});
