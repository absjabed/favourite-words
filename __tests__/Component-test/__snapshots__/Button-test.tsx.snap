// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#555CC4",
      "borderRadius": 7,
      "color": "#ffffff",
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 16,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontFamily": "Ubuntu-Regular",
      }
    }
  />
</View>
`;
